tensorflow/src/main/scala/magnolify/tensorflow/ExampleType.scala

```diff
@@ -119,30 +120,23 @@ object ExampleField {
         f
       }
 
-    override protected def buildSchema(k: String)(cm: CaseMapper): Schema = {
-      val sb = Schema
-        .newBuilder()
+    override protected def buildSchema(cm: CaseMapper): Schema = {
+      val sb = Schema.newBuilder()
       getDoc(caseClass.annotations, caseClass.typeName.full).foreach(sb.setAnnotation)
-
       caseClass.parameters.foldLeft(sb) { (b, p) =>
-        val s = p.typeclass.buildSchema(key(k, cm.map(p.label)))(cm)
-        val annotatedFs = s.getFeatureList.asScala
-          .map { f =>
-            val maybeAnnotation = if (f.hasAnnotation) {
-              Some(f.getAnnotation)
-            } else {
-              getDoc(p.annotations, s"${caseClass.typeName.full}#${key(k, cm.map(p.label))}")
-            }
-
-            maybeAnnotation match {
-              case Some(a) => f.toBuilder.setAnnotation(a).build()
-              case None    => f
-            }
-          }
-        b.addAllFeature(annotatedFs.asJava)
+        val n = cm.map(p.label)
+        val s = p.typeclass.schema(cm)
+        val fs = s.getFeatureList.asScala.map { f =>
+          val fb = f.toBuilder
+          val k = if (f.hasName) key(n, f.getName) else n // primitives do not have name
```

Comment:
primitives don't have a name ... but you call `setName` on `k` below?

Reply:
I mean the primitive do not have a name in their schema. Eg you can't give a name to the `Primitive[Long]`.

If the field is a primitive, we set the field name alone: `n`.
If the field is a nested record, we prepend the name `n` to the nested one `f.getName`.